packages/klighd-core/src/sidebar/sidebar-panel-registry.ts

```diff
@@ -49,7 +49,7 @@ export class SidebarPanelRegistry extends Registry {
             if (this._currentPanelID === action.id && action.state === "show") return;
             if (this._currentPanelID !== action.id && action.state === "hide") return;
 
-            if (this._currentPanelID === action.id) {
+            if (this._currentPanelID === action.id && !(this.currentPanel as SidebarPanel).panelPinned) {
```

Comment:
This seems wrong to dispatch a ToggleSidebarAction that does not toggle the sidebar. Does something speak against not dispatching the action in the first place if the panel is pinned in all cases where it could be toggled?